src/main/java/algorithms/curated170/medium/BoundaryOfBinaryTree.java

```diff
@@ -0,0 +1,59 @@
+package algorithms.curated170.medium;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import algorithms.datastructures.TreeNode;
+
+public class BoundaryOfBinaryTree {
+    public List<Integer> boundaryOfBinaryTree(TreeNode root) {
+        List<Integer> boundary = new ArrayList<>();
+        if (root == null) {
+            return boundary;
+        }
+        boundary.add(root.val);
+
+        getLeftBorder(root.left, boundary);
+        if (root.left != null || root.right != null) {
+            getLeaves(root, boundary);
+        }
+        getRightBorder(root.right, boundary);
```

Comment:
```suggestion
         if (root.left != null || root.right != null) {
             getLeftBorder(root.left, boundary);
             getLeaves(root, boundary);
             getRightBorder(root.right, boundary);
        }
```

![image](https://user-images.githubusercontent.com/60903744/124368480-14e30b00-dc6a-11eb-81da-e37de164de02.png)